include/kernel.h

```diff
@@ -5062,6 +5095,28 @@ __syscall void k_str_out(char *c, size_t n);
  */
 __syscall int k_float_disable(struct k_thread *thread);
 
+#ifdef CONFIG_THREAD_RUNTIME_STATS
+
+/**
+ * @brief Get the runtime statistics of a thread
+ *
+ * @param thread ID of thread.
+ * @param stats Pointer to struct to copy statistics into.
+ * @return -EINVAL if null pointers, otherwise 0
+ */
+int k_thread_runtime_stats_get(struct k_thread *thread,
```

Comment:
use `k_tid_t` here instead of `struct k_thread *` for application-facing APIs that take thread references.

(yes there is cleanup elsewhere in the kernel with respect to this)

Reply:
Changed to use `k_tid_t`.